### PR TITLE
improve parsing for stratified parameters

### PR DESCRIPTION
**Pull request by mwdchang**
### Summary
Improve parser to reduce the number of false positives where we incorrectly identify something as stratified but it is not. Use the context (grounding info) data, if applicable to determine whether a parameter is the result of an earlier stratification process.

### Testing
Load and view the attached AMR model 

[SEIRHD.vacc.model.for.LA.County.t0.10_28_2021.1.json](https://github.com/user-attachments/files/17153595/SEIRHD.vacc.model.for.LA.County.t0.10_28_2021.1.json)

parameters like `NPI_mult` and `vacc_mult` will denoted as stand-alone parameters and not grouped under a stratified category. Ditto for the`r_*` variables.

Closes: https://github.com/DARPA-ASKEM/terarium/issues/4764

### CODE REVIEW

File: packages/client/hmi-client/src/model-representation/mira/mira.ts [hunk 1]
```diff
@@ -65,20 +67,22 @@ export const getContextKeys = (miraModel: MiraModel) => {
...
 				Object.entries(miraConcept.context).forEach(([k, v]) => {
 					addWithGuard(k, v as string);
 				});
 			});
 		}
 	});
 
-	const modifiers = [...modifierKeys];
-	return modifiers;
+	// const modifiers = [...modifierKeys];
```
Comment (YohannParis) [inline packages/client/hmi-client/src/model-representation/mira/mira.ts:77]:
```suggestion
```

File: packages/client/hmi-client/src/model-representation/mira/mira.ts [hunk 2]
```diff
@@ -107,14 +112,27 @@ export const collapseParameters = (miraModel: MiraModel, miraTemplateParams: Mir
 			continue;
 		}
 
+		const nameContainsContext = _.intersection(tokens, contextValues).length > 0;
+		const hasNumericSuffixes =
+			tokens.length > 1 &&
+			tokens.every((d, idx) => {
```
Comment (YohannParis) [inline packages/client/hmi-client/src/model-representation/mira/mira.ts:118]: Why the `tokens.length > 1` test, `every` works on empty array as well. Or just to be sure?

File: packages/client/hmi-client/src/model-representation/mira/mira.ts [hunk 3]
```diff
@@ -33,8 +33,9 @@ export const emptyMiraModel = () => {
 /**
  * Collection of MMT related functions
  * */
-export const getContextKeys = (miraModel: MiraModel) => {
```
Comment (YohannParis) [inline packages/client/hmi-client/src/model-representation/mira/mira.ts:36]: You could keep this function with the following, so you do not have to modifed the service. Could still be useful
```javascript
export const getContextKeys = (miraModel: MiraModel) => getContext(miralModel).keys;
```

**Review by YohannParis — APPROVED** [review]
LGTM

File: packages/client/hmi-client/src/model-representation/mira/mira.ts (hunk 2 above)
Comment (mwdchang) [reply packages/client/hmi-client/src/model-representation/mira/mira.ts:117]: We want `x_123` to be true, but `x_abc` to be false and `x` to be false.

**Review by blanchco — APPROVED** [review]
tested and looks good

**Pull request merged**